src/bootstrap/src/bin/main.rs

```diff
@@ -15,8 +15,20 @@ use bootstrap::{
     human_readable_changes, t,
 };
 use build_helper::ci::CiEnv;
-
+// These are conditionally enabled if the user specifies `BOOTSTRAP_LOG=...`, which builds bootstrap
+// with `--features=logging`.
+#[cfg(feature = "logging")]
+use tracing::*;
+#[cfg(feature = "logging")]
+use tracing_subscriber::EnvFilter;
+#[cfg(feature = "logging")]
+use tracing_subscriber::prelude::*;
+
+#[cfg_attr(feature = "logging", instrument(level = "trace", name = "main"))]
 fn main() {
+    #[cfg(feature = "logging")]
+    setup_logging();
```

Comment:
Again, all these log names will cause a lot of confusion eventually

Reply:
Fixed by changing to `tracing` feature and renaming to `setup_tracing`.